instance_dire_maul.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+This placeholder for the instance is needed for dungeon finding to be able
+to give credit after the boss defined in lastEncounterDungeon is killed.
+Without it, the party doing random dungeon won't get satchel of spoils and
+gets instead the deserter debuff.
+*/
+
+#include "GameObject.h"
+#include "InstanceScript.h"
+#include "Map.h"
+#include "ObjectAccessor.h"
+#include "ScriptedCreature.h"
+#include "ScriptMgr.h"
+#include "diremaul.h"
+
+// Bosses (East)
+// 0 - Pusillin
+// 1 - Lethtendris
+// 2 - Hydrospawn
+// 3 - Zevrim Thornhoof
+// 4 - Alzzin the Wildshaper
+
+// West
+// 5 - Tendris Warpwood
+// 6 - Magister Kalendris
+// 7 - Tsu'zee
+// 8 - Illyanna Ravenoak
+// 9 - Immol'thar
+// 10 - Prince Tortheldrin
+// 17 - CRYSTAL_01
+// 18 - CRYSTAL_02
+// 19 - CRYSTAL_03
+// 20 - CRYSTAL_04
+// 21 - CRYSTAL_05
+// 22 - FORCEFIELD
+
+// North
+// 11 - Guard Mol'dar
+// 12 - Stomper Kreeg
+// 13 - Guard Fengus
+// 14 - Guard Slip'kik
+// 15 - Captain Kromcrush
+// 16 - King Gordok
+
+uint8 const EncounterCount = 23;
+
+uint32 const CrystalMobs[2] = { NPC_ARCANE_ABERRATION, NPC_MANA_REMNANT };
+
+enum Events
+{
+    EVENT_CRYSTAL_CREATURE_STORE                = 1,
+    EVENT_CRYSTAL_CREATURE_CHECK                = 2,
+    EVENT_IMMO_DEAD_CHECK                       = 3
+};
+
+class instance_dire_maul : public InstanceMapScript
+{
+public:
+    instance_dire_maul() : InstanceMapScript("instance_dire_maul", 429) { }
+
+    struct instance_dire_maul_InstanceMapScript : public InstanceScript
+    {
+        instance_dire_maul_InstanceMapScript(InstanceMap* map) : InstanceScript(map)
+        {
+            SetHeaders(DataHeader);
+            SetBossNumber(EncounterCount);
+        }
+
+        void OnCreatureCreate(Creature* creature) override
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_IMMOLTHAR:
+                    _immoGUID = creature->GetGUID();
+                    // we make Immolthar non attackable, otherwise players with pets can pull him out of the forcefield
+                    // TODO: this change isnt correct but since of today (13.09.2020) mmaps dont support doors

```

Comment:
was this code copied from somewhere ?

Reply:
I don't believe so, that's the original code from the unchanged file.